tests/components/igloohome/conftest.py

```diff
@@ -0,0 +1,72 @@
+"""Common fixtures for the igloohome tests."""
+
+from collections.abc import Generator
+from unittest.mock import AsyncMock, patch
+
+from igloohome_api import GetDeviceInfoResponse, GetDevicesResponse
+import pytest
+
+from homeassistant.components.igloohome.const import DOMAIN
+from homeassistant.const import CONF_CLIENT_ID, CONF_CLIENT_SECRET
+from homeassistant.core import HomeAssistant
+
+from tests.common import MockConfigEntry
+
+GET_DEVICE_INFO_RESPONSE_LOCK = GetDeviceInfoResponse(
+    id="123456",
+    type="Lock",
+    deviceId="OE1X123cbb11",
+    deviceName="Front Door",
+    pairedAt="2024-11-09T11:19:25+00:00",
+    homeId=[],
+    linkedDevices=[],
+    batteryLevel=100,
+)
+
+
+@pytest.fixture
+def mock_setup_entry() -> Generator[AsyncMock]:
+    """Override async_setup_entry."""
+    with patch(
+        "homeassistant.components.igloohome.async_setup_entry", return_value=True
+    ) as mock_setup_entry:
+        yield mock_setup_entry
+
+
+@pytest.fixture
+async def mock_auth() -> Generator[AsyncMock]:
+    """Set up the mock for the igloohome_api Auth module. Defaults to always successfully operate."""
+    with patch(
+        "igloohome_api.Auth.async_get_access_token",
```

Comment:
Please patch this at the places where we use this, so `homeassistant.components.igloohome.....`

Reply:
Sure thing. Curious question: Is this a common practice for python/HA? As it would seem cleaner (code-wise) to patch the library directly so that all future usages are also automatically patched rather than targeting the individual modules.

Reply:
https://docs.python.org/3/library/unittest.mock.html#where-to-patch

We also just consider it a good practice to patch where we use it, this way we reduce the risk of side effects. There are integrations which share a lib for example

Reply:
Thanks a bunch for the link and clarification!